sigopt/xgboost/experiment.py

```diff
@@ -3,50 +3,107 @@
 import copy
 
 from .. import create_experiment
+from .constants import (
+  SEARCH_BOUNDS,
+  SEARCH_PARAMS,
+  DEFAULT_SEARCH_PARAMS,
+  DEFAULT_BO_ITERATIONS,
+  DEFAULT_NUM_BOOST_ROUND,
+  DEFAULT_REGRESSION_METRIC,
+  DEFAULT_CLASSIFICATION_METRIC,
+  REGRESSION_METRIC_CHOICES,
+  CLASSIFICATION_METRIC_CHOICES,
+  SUPPORTED_METRICS_TO_OPTIMIZE
+)
+
+def check_experiment_config(experiment_config):
+  pass
 
-DEFAULT_CLASSIFICATION_METRICS = ['accuracy', 'precision', 'recall', 'F1']
-DEFAULT_REGRESSION_METRICS = ['mean absolute error', 'mean squared error']
-DEFAULT_NUM_BOOST_ROUND = 10
 
 class XGBExperiment:
   def __init__(self, experiment_config, dtrain, evals, params, num_boost_round, run_options):
-    self.experiment_config = experiment_config
+    self.experiment_config_parsed = copy.deepcopy(experiment_config)
     self.dtrain = dtrain
     self.evals = evals
     self.params = params
     self.num_boost_round = num_boost_round
     self.run_options = run_options
     self.sigopt_experiment = None
 
-  def parse_and_create_parameter_space(self):
-    # TODO: deal with later
-    pass
-
-  def parse_and_create_experiment(self):
-    experiment_config_parsed = copy.deepcopy(self.experiment_config)
+  def parse_and_create_metrics(self):
+    if 'metrics' in self.experiment_config_parsed and isinstance(self.experiment_config_parsed['metrics'], list):
+      pass  # do nothing
+    else:
+      if 'metrics' not in self.experiment_config_parsed:
+        # pick a default metric
+        if 'objective' in self.params:
+          objective = self.params['objective']
+          if objective.split(':')[0] in ['binary', 'multi']:
+            metric_to_optimize = DEFAULT_CLASSIFICATION_METRIC
+          else:
+            metric_to_optimize = DEFAULT_REGRESSION_METRIC  # do regression if anything else (including ranking)
+        else:
+          metric_to_optimize = DEFAULT_REGRESSION_METRIC
+      else:
+        assert self.experiment_config_parsed['metrics'] in SUPPORTED_METRICS_TO_OPTIMIZE
+        metric_to_optimize = self.experiment_config_parsed['metrics']
+      self.experiment_config_parsed['metrics'] = [{
+        'name': metric_to_optimize,
+        'strategy': 'optimize',
+        'objective': 'maximize'
+      }]
 
     # Check experiment config optimization metric
-    for metric in experiment_config_parsed['metrics']:
+    for metric in self.experiment_config_parsed['metrics']:
       if metric['strategy'] == 'optimize':
-        assert metric['name'] in DEFAULT_CLASSIFICATION_METRICS or metric['name'] in DEFAULT_REGRESSION_METRICS
+        assert metric['name'] in CLASSIFICATION_METRIC_CHOICES or metric['name'] in REGRESSION_METRIC_CHOICES
```

Comment:
`assert self.experiment_config_parsed['metrics'] in SUPPORTED_METRICS_TO_OPTIMIZE`?